### ISSUE TITLE
Standardize release notes

### ISSUE THREAD

**Issue by dblock** [opensearch-project/.github#39]
Standardize release notes across all products that we release in this org.

- Should one include every PR or remove some?
- ...

**Comment by dblock** [comment]
@ahopp

**Comment by ahopp** [comment]
**RE: tracking all PRs**, I'm happy to defer here but my instinct it the release notes wouldn't need to include all PRs but I would ere on the side of caution until we get more feedback. My experience is release notes are meant for users and the community with the intent to provide a quick guide to what meaningful changes occurred - not an exhaustive list of all the work that was completed (github will store that for us). @dblock @elfisher @stockholmux thoughts? 

**RE: humanly readable**, yes, I think we should seek to make the release notes as informative and functional as possible. At the end of the day this content is meant to be educational so making release notes legible to end users and developers without deep diving the associated PR is important. Some simple guidelines for this would be; 

1. Use plain language. Explain the feature like you would to a real person. Avoid Jargon. 
2. Be as concise as possible. It shouldn’t take more than a sentence or two to explain it at the high-level. 
3. Explain why it matters / Show value. Don't just list what was worked on. 
4. Keep them logically grouped e.g. new feature, enhancements, bugs, etc. 

**RE: the standard format**, I'm not there is a signal format given the different information we would want to convey (e.g. across features, bug fixes, or enhancements) but I can provide some templates that might cover most cases. A few examples that I would consider good release notes templates;

- We added [X] so you can now do [Y]. #PR
- Improved [X] to allow better [Y]. #PR
- Removed [X] to improve [Y]. #PR
- You can now do [X] instead of [Y]. #PR
- [Z] no longer does [Y]. This means you no longer need to do [X]. #PR

**RE: categories**, I think the current template is fine for sections e.g [here](https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md)

I think @peternied might have provided general guidance in the past. Anything I missed @peternied?

**Comment by stockholmux** [comment]
@ahopp I have to disagree with a point in the standard format. "_**We** added [X] so you can now do [Y]. #PR_" 

"We" is super non-specific (Who: You and Frankenstein?) and not useful. I think just starting with "Added..." is a better way to go and more consistent with the subsequent ones.

**Comment by elfisher** [comment]
@ahopp I like the idea. Added X so you can do Y is clear. So are the others. That should help make the release notes more readable.
